test/e2e/specs/editor/various/block-switcher.spec.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	blockSwitcher: async ( { page }, use ) => {
+		await use( new blockSwitcher( { page } ) );
+	},
+} );
+
+test.describe( 'Block Switcher', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should show the expected block transforms on the list block when the blocks are removed', async ( {
+		page,
+		blockSwitcher,
+		pageUtils,
+	} ) => {
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher exists.
+		expect( page.locator( 'role=button[name="List"i]' ) ).toBeTruthy();
+
+		// Verify the correct block transforms appear.
+		expect( await blockSwitcher.getAvailableBlockTransforms() ).toEqual(
+			expect.arrayContaining( [
+				'Group',
+				'Paragraph',
+				'Heading',
+				'Quote',
+				'Columns',
+			] )
+		);
+	} );
+
+	test( 'Should show the expected block transforms on the list block when the quote block is removed', async ( {
+		page,
+		blockSwitcher,
+		pageUtils,
+	} ) => {
+		const wp = '';
+
+		// Remove the quote block from the list of registered blocks.
+		await page.evaluate( () => {
+			wp.blocks.unregisterBlockType( 'core/quote' );
+		} );
+
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher exists.
+		expect( page.locator( 'role=button[name="List"i]' ) ).toBeTruthy();
+
+		// Verify the correct block transforms appear.
+		expect( await blockSwitcher.getAvailableBlockTransforms() ).toEqual(
+			expect.arrayContaining( [
+				'Group',
+				'Paragraph',
+				'Heading',
+				'Columns',
+			] )
+		);
+	} );
+
+	test( 'Should not show the block switcher if all the blocks the list block transforms into are removed', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		const wp = '';
+		async function getAvailableBlockTransforms() {
+			return page.evaluate( ( buttonSelector ) => {
+				return Array.from(
+					document.querySelectorAll( buttonSelector )
+				).map( ( button ) => {
+					return button.textContent;
+				} );
+			}, '.block-editor-block-switcher__popover .block-editor-block-switcher__transforms__menugroup button' );
+		}
+
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Remove the paragraph and quote block from the list of registered blocks.
+		await page.evaluate( () => {
```

Comment:
This is executed before the key press in the original test. I think we should follow the original behavior :).